docs/modules/vector-search/pages/vector-collections.adoc

```diff
@@ -0,0 +1,498 @@
+VectorDocument<String> document = VectorDocument.of("document value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+CompletionStage<VectorDocument<String>> result = vectorCollection.putAsync("1", document);
+----
+--
+====
+
+To put several documents to a vector collection, use the method `putAllAsync` of the `VectorCollection` class.
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+VectorDocument<String> document1 = VectorDocument.of("some document1 value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+VectorDocument<String> document2 = VectorDocument.of("some document2 value", VectorValues.of(new float[] {1.2f, -0.3f, 2.2f}));
+CompletionStage<Void> result = vectorCollection.putAllAsync(Map.of("1", document1, "2", document2));
+----
+--
+====
+
+=== Read objects
+To get document from a vector collection, use the method `getAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<VectorDocument<String>> result = vectorCollection.getAsync("1");
+----
+--
+====
+
+=== Update objects
+To update single document in a vector collection, use the method `putAsync` or `setAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+VectorDocument<String> document = VectorDocument.of("some document value", VectorValues.of(new float[] {0.2f, 0.9f, -1.2f}));
+CompletionStage<Void> result = vectorCollection.setAsync("1", document);
+----
+--
+====
+
+=== Delete objects
+To delete document from a vector collection, use the method `deleteAsync` or `removeAsync` of the `VectorCollection` class.
+
+[tabs]
+====
+Java::
++
+--
+[source,java]
+----
+VectorCollection<String, String> vectorCollection = VectorCollection.getCollection(hazelcastInstance, collectionName);
+CompletionStage<Void> resultDelete = vectorCollection.deleteAsync("1");
+CompletionStage<VectorDocument<String>> resultRemove = vectorCollection.removeAsync("2");
+----
+--
+====
+
+NOTE: These methods do not delete index nodes but mark them as deleted. This may impact search speed and memory usage. To permanently remove nodes from the index, you must run index optimization after deletion. See <<optimize-collection, optimize method>>
+
+== Similarity search
+
+Vector search returns objects with vectors that are most similar to that of the query, based on specified metrics.
+When submitting a search query, you have the option to include one vector values representing your query embeddings and specify a limit on the number of results to retrieve.
```

Comment:
> include one vector values representing your query embeddings

this is hard to understand, but it's hard to describe `isIncludeVectors` concisely

Reply:
b2fd6c714396ed311c966efd372253f1d485ff18